imap-proto/src/body.rs

```diff
@@ -15,6 +15,11 @@ named!(pub section_part<Vec<u32>>, do_parse!(
 ));
 
 named!(pub section_msgtext<MessageSection>, alt!(
+    do_parse!(tag_no_case!("HEADER.FIELDS") >>
```

Comment:
I'd prefer to have the immediate elements of `do_parse!()` indented at one level, rather than this way.

(Of course this PR is also still failing the tests.)